pkg/golinters/goanalysis/runners.go

```diff
@@ -42,14 +43,32 @@ func runAnalyzers(cfg runAnalyzersConfig, lintCtx *linter.Context) ([]result.Iss
 		pkgs = lintCtx.OriginalPackages
 	}
 
+	pkgByPath := make(map[string]*packages.Package, len(pkgs))
+	for _, pkg := range pkgs {
+		pkgByPath[pkg.PkgPath] = pkg
+	}
+
 	issues, pkgsFromCache := loadIssuesFromCache(pkgs, lintCtx, cfg.getAnalyzers())
+
 	var pkgsToAnalyze []*packages.Package
 	for _, pkg := range pkgs {
 		if !pkgsFromCache[pkg] {
 			pkgsToAnalyze = append(pkgsToAnalyze, pkg)
+
+			// Also add the local packages imported by a package to analyze.
+			// Some linters produce reports on a package reported by another one.
+			// This is only needed for local imports.
+			for _, v := range pkg.Imports {
+				if p, found := pkgByPath[v.PkgPath]; found {
+					pkgsToAnalyze = append(pkgsToAnalyze, p)
+				}
+			}
 		}
 	}
 
+	// keep only unique packages
+	pkgsToAnalyze = slices.Compact(pkgsToAnalyze)
```

Comment:
This line won't remove duplicated packages. It's needed to sort the slice first (see https://boyter.org/posts/deduplicate-slice-go-sort-or-map/).

We can add `log` to the function to proof:
1. The current implementation in the PR `slices.Sort` has duplicates:
```
WARN [linters_context] Packages before 'keep only unique': [reproduction/logger reproduction/somepackage reproduction/logger reproduction reproduction/somepackage] 
WARN [linters_context] Packages after 'keep only unique': [reproduction/logger reproduction/somepackage reproduction/logger reproduction reproduction/somepackage] 
```
2. But after I add `slices.Sort` the `pkgsToAnalyze` has unique values:
```
WARN [linters_context] Packages before 'keep only unique': [reproduction/logger reproduction/somepackage reproduction/logger reproduction reproduction/somepackage] 
WARN [linters_context] Packages after 'keep only unique': [reproduction reproduction/logger reproduction/somepackage]
```